src/consumer.rs

```diff
@@ -173,54 +165,129 @@ impl JSValue<'_> for OffsetWrapper {
-        response.set_property(ERROR_CODE_KEY, error_code)?;
-        response.set_property(HIGH_WATERMARK_KEY, high_watermark)?;
-        response.set_property(LAST_STABLE_OFFSET_KEY, last_stable_offset)?;
-        response.set_property(LOG_START_OFFSET_KEY, log_start_offset)?;
-
-        if let Some(transactions) = self.0.aborted {
-            let aborted = js_env.create_array_with_len(transactions.len())?;
-            for (index, transaction) in transactions.into_iter().enumerate() {
-                let mut tx = JsObject::create(js_env)?;
-                let producer_id = js_env.create_int64(transaction.producer_id)?;
-                let first_offset = js_env.create_int64(transaction.first_offset)?;
-                tx.set_property(PRODUCER_ID_KEY, producer_id)?;
-                tx.set_property(FIRST_OFFSET_KEY, first_offset)?;
-                let element = tx.try_to_js(js_env)?;
-
-                // Update the array of aborted transactions;
-                js_env.set_element(aborted, element, index)?;
-            }
+#[node_bindgen]
+impl FetchablePartitionResponseWrapper {
+    #[node_bindgen(constructor)]
+    fn new() -> Self {
+        Self(None)
+    }
+    fn set_inner(&mut self, inner: Option<FetchablePartitionResponse<RecordSet>>) {
+        self.0 = inner;
+    }
+    #[node_bindgen(getter)]
+    fn error_code(&self) -> Option<i32> {
+        Some(self.0.as_ref()?.error_code as i32)
+    }
+
+    #[node_bindgen(getter)]
+    fn partition_index(&self) -> Option<i32> {
+        Some(self.0.as_ref()?.partition_index)
+    }
+
+    #[node_bindgen(getter)]
+    fn high_watermark(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.high_watermark)
+    }
+
+    #[node_bindgen(getter)]
+    fn last_stable_offset(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.last_stable_offset)
+    }
+
+    #[node_bindgen(getter)]
+    fn log_start_offset(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.log_start_offset)
+    }
+    #[node_bindgen(getter)]
+    fn aborted(&'static self) -> Vec<AbortedTransactionWrapper> {
+        let mut aborted_transactions: Vec<AbortedTransactionWrapper> = Vec::new();
+
+        let inner = if let Some(inner) = self.0.as_ref() {
+            inner
+        } else {
+            return Vec::new();
+        };
 
-            // set the aborted transactions;
-            response.set_property(ABORTED_KEY, aborted)?;
+        if let Some(ref transactions) = inner.aborted {
+            for i in transactions {
+                aborted_transactions.push(AbortedTransactionWrapper(Some(&i)));
+            }
         }
+        aborted_transactions
+    }
 
-        let record_set = RecordSetWrapper(self.0.records).try_to_js(js_env)?;
+    #[node_bindgen(getter)]
+    fn records(&'static self) -> Option<RecordSetWrapper> {
+        Some(RecordSetWrapper(&self.0.as_ref()?.records))
+    }
+    #[node_bindgen]
+    fn to_records(&'static self) -> Vec<String> {
```

Comment:
I'm seeing a bunch of this `&'static self` which smells really fishy to me. Why does this have a `'static` lifetime? How is this method used, and how is the underlying object created? I'm having a very difficult time imagining that this lifetime is actually correct. If the object being referenced is not really truly static (as in `STATIC` or `lazy_static!`) then I think this is a recipe for memory errors.

Could you walk me through the intended usage of this API?